torch/_meta_registrations.py

```diff
@@ -143,3 +132,86 @@ def meta_repeat_interleave_Tensor(repeats, output_size=None):
             "cannot repeat_interleave a meta tensor without output_size"
         )
     return repeats.new_empty(output_size)
+
+@torch.library.impl(meta_lib, "index.Tensor")
+def meta_index_Tensor(self, indices):
+    # aten::index is the internal advanced indexing implementation
+    check(len(indices) <= self.ndim, lambda: f"too many indices for tensor of dimension {self.ndim} (got {len(indices)})")
+    # checkIndexTensorTypes
+    result = []
+    for i, index in enumerate(indices):
+        if index is not None:
+            check(
+                index.dtype in [torch.long, torch.int8, torch.bool],
+                lambda: "tensors used as indices must be long, byte or bool tensors"
+            )
+            if index.dtype in [torch.int8, torch.bool]:
+                nonzero = index.nonzero()
+                for j in range(index.ndim):
+                    result.append(nonzero.select(1, j))
+            else:
+                result.append(index)
+        else:
+            result.append(index)
+    indices = result
+    # expand_outplace
+    import torch._refs as refs  # avoid import cycle in mypy
+    indices = list(refs._maybe_broadcast(*indices))
+    # add missing null tensors
+    while len(indices) < self.ndim:
+        indices.append(None)
+
+    # hasContiguousSubspace
+    #   true if all non-null tensors are adjacent
+    # See:
+    # https://stackoverflow.com/questions/53841497/why-does-numpy-mixed-basic-advanced-indexing-depend-on-slice-adjacency
```

Comment:
A better reference for this may be https://numpy.org/doc/stable/user/basics.indexing.html#combining-advanced-and-basic-indexing= (or a link to a pinned version of that page).